mirdata/giantsteps_key.py

```diff
@@ -0,0 +1,368 @@
+description of the properties of this dataset can be found:
+
+Ángel Faraldo (2017). Tonality Estimation in Electronic Dance Music: A Computational and Musically Informed Examination.
+PhD Thesis. Universitat Pompeu Fabra, Barcelona.
+
+This dataset is mainly intended to assess the performance of computational key estimation algorithms in electronic dance
+music subgenres.
+
+All the data of this dataset is licensed with Creative Commons Attribution Share Alike 4.0 International.
+
+"""
+
+import csv
+import json
+import zipfile
+
+import librosa
+import numpy as np
+import os
+
+from mirdata import download_utils
+from mirdata import jams_utils
+from mirdata import track
+from mirdata import utils
+
+DATASET_DIR = 'GiantSteps_key'
+REMOTES = {
+    'audio': download_utils.RemoteFileMetadata(
+        filename='audio.zip',
+        url='https://zenodo.org/record/1095691/files/audio.zip?download=1',
+        checksum='8ec9ade888d5a88ce435d7fda031929b',
+        destination_dir='.',
+    ),
+    'keys': download_utils.RemoteFileMetadata(
+        filename='keys.zip',
+        url='https://zenodo.org/record/1095691/files/keys.zip?download=1',
+        checksum='775b7d17e009f5818544cf505b6a96fd',
+        destination_dir='.',
+    ),
+    'metadata': download_utils.RemoteFileMetadata(
+        filename='original_metadata.zip',
+        url='https://zenodo.org/record/1095691/files/original_metadata.zip?download=1',
+        checksum='54181e0f34c35d9720439750d0b08091',
+        destination_dir='.',
+    ),
+}
+
+DATA = utils.LargeData('giantsteps_key_index.json')
+
+
+class Track(track.Track):
+    """giantsteps_key track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        audio_path (str): track audio path
+        keys_path (str): key annotation path
+        metadata_path (str): sections annotation path
+        title (str): title of the track
+        track_id (str): track id
+
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError(
+                '{} is not a valid track ID in giantsteps_key'.format(track_id)
+            )
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+        self.audio_path = os.path.join(self._data_home, self._track_paths['audio'][0])
+        self.keys_path = os.path.join(self._data_home, self._track_paths['key'][0])
+        self.metadata_path = (
+            os.path.join(self._data_home, self._track_paths['meta'][0])
+            if self._track_paths['meta'][0] is not None
+            else None
+        )
```

Comment:
suggestion for small efficiency boost:
add this here:
`self._metadata = None`

and see comments below~